src/server/lib/idapi/invertOptOutConsents.ts

```diff
@@ -0,0 +1,29 @@
+import { Consent } from '@/shared/model/Consent';
+
+/**
+ * See PR https://github.com/guardian/identity/pull/2079 for more details why this is needed.
+ * This is horrible but necessary due to business requirements because we cannot change
+ * the consent IDs "eg. Profiling_Optout" on the server data model or downstream integrations
+ */
+
+type ConversionString = '_optout' | '_optin';
+
+const invertConsents =
+  (input: ConversionString, output: ConversionString) =>
+  (pageConsents: Consent[]) => {
```

Comment:
A minor point, is it worth using curly braces and an explicit `return` here? I has to double check it was a function returning a function ...